src/Tasks/CreateItem.cs

```diff
@@ -126,46 +132,97 @@ private List<ITaskItem> CreateOutputItems(Dictionary<string, string> metadataTab
             return outputItems;
         }
 
+        /// <summary>
+        /// Attempts to expand wildcards and logs warnings or errors for attempted drive enumeration.
+        /// </summary>
+        private bool TryExpandingWildcards(ITaskItem[] expand, bool includeAttribute)
+        {
+            const string CreateItemTask = "CreateItem";
+            string attributeType;
+            string fileSpec;
+
+            FileMatcher.SearchAction searchAction;
+
+            if (includeAttribute)
+            {
+                (Include, searchAction, fileSpec) = ExpandWildcards(expand);
+                attributeType = XMakeAttributes.include;
+            }
+            else
+            {
+                (Exclude, searchAction, fileSpec) = ExpandWildcards(expand);
+                attributeType = XMakeAttributes.exclude;
+            }
+
+            // Log potential drive enumeration glob anomalies when applicable.
+            if (searchAction == FileMatcher.SearchAction.ReturnLogDriveEnumerationWildcard)
+            {
+                Log.LogWarningWithCodeFromResources("WildcardResultsInDriveEnumeration", EscapingUtilities.UnescapeAll(fileSpec), attributeType, CreateItemTask);
+            }
+            else if (searchAction == FileMatcher.SearchAction.ReturnThrowOnDriveEnumerationWildcard)
+            {
+                Log.LogErrorWithCodeFromResources("WildcardResultsInDriveEnumeration", EscapingUtilities.UnescapeAll(fileSpec), attributeType, CreateItemTask);
+                return false;
+            }
+
+            return true;
```

Comment:
I think it's a little better to return !Log.HasLoggedErrors – more future-proof.